src/Config/ChainFileLocator.php

```diff
@@ -63,18 +66,23 @@ public function locate($name, $currentPath = null, $first = false)
     }
 
     /**
-     * Reverses the locator order so that higher priority locators overwrite lower priority ones.
+     * Adds new files to existing array without overwriting existing keys, except if they are numeric.
      *
-     * @param bool $first Whether to return the first occurrence or an array of filenames
+     * @param array $newFiles The files to be added
+     * @param array $allFiles The existing files
      *
-     * @return FileLocatorInterface[] The locators array
+     * @return array Merged list of files
      */
-    private function getLocators($first)
+    private function mergeFiles(array $newFiles, array $allFiles)
     {
-        if (false === $first) {
-            return array_reverse($this->locators);
+        foreach ($newFiles as $k => $v) {
+            if (is_numeric($k)) {
+                $allFiles[] = $v;
+            } elseif (!isset($allFiles[$k])) {
```

Comment:
This means the locator with the highest priority wins, right? Because it is the first one that gets to set non-numeric entries.